cql3/statements/create_keyspace_statement.cc

```diff
@@ -170,14 +172,14 @@ std::optional<sstring> check_restricted_replication_strategy(
         case db::tri_mode_restriction_t::mode::FALSE:
             // Scylla was configured to allow SimpleStrategy, but let's warn
             // if it's used on a cluster which *already* has multiple DCs:
-            if (qp.proxy().get_token_metadata_ptr()->get_topology().get_datacenter_endpoints().size() > 1) {
+            if (topology.get_datacenter_endpoints().size() > 1) {
```

Comment:
I know it is a nitpick however, there is a function better suited for this, `topology.get_datacenters` since you already refactored this code, it might be beneficial to make it more readable.